--- /tmp/pristine_ripemd160.cpp	2026-07-24 07:34:05.650766599 -0600
+++ hash/ripemd160.cpp	2026-07-24 07:09:47.044305054 -0600
@@ -37,10 +37,16 @@
 }
 
 #ifndef WIN64
+#if defined(__x86_64__) || defined(__i386__)
 inline uint32_t _rotl(uint32_t x, uint8_t r) {
   asm("roll %1,%0" : "+r" (x) : "c" (r));
   return x;
 }
+#else
+inline uint32_t _rotl(uint32_t x, uint8_t r) {
+  return (x << r) | (x >> (32 - r));
+}
+#endif
 #endif
 
 #define ROL(x,n) _rotl(x,n)
