--- /tmp/pristine_sha256.cpp	2026-07-24 07:34:03.834763797 -0600
+++ hash/sha256.cpp	2026-07-24 07:09:26.028249461 -0600
@@ -34,10 +34,16 @@
 #ifndef WIN64
 #define _byteswap_ulong __builtin_bswap32
 #define _byteswap_uint64 __builtin_bswap64
+#if defined(__x86_64__) || defined(__i386__)
 inline uint32_t _rotr(uint32_t x, uint8_t r) {
   asm("rorl %1,%0" : "+r" (x) : "c" (r));
   return x;
 }
+#else
+inline uint32_t _rotr(uint32_t x, uint8_t r) {
+  return (x >> r) | (x << (32 - r));
+}
+#endif
 #endif
 
 #define ROR(x,n) _rotr(x, n)
